### ISSUE TITLE
 The visual presentation of text and images of text has a contrast ratio of at least 4.5:1

### ISSUE THREAD

**Issue by ecureuill** [solex2006/SELIProject#92]
This is a "not-end" requirement. What I want to mean is that this should be considered for any new page you create that Student can access. Also should correct the old pages.

I will use the label **Feature Design Notes** to this cases
 
***************
## Scenario:  No text color nor background
- [ ] Not specifying background color, not specifying text color, and not using technology features that change those defaults

With this technique the author avoids having to do any contrast measures by simply not specifying the text color and not specifying the background. As a result the colors of both are completely determined by the user agent. Some people who have vision disabilities set their user agent to override certain colors that they have trouble seeing. This technique will help avoid a situation where the user agent and web site conflict with each other over the foreground and/or background colors resulting in the same color for text and background, which would make it invisible for user who set their own colors in their browser or Assistive Technology.

### Tests Procedure
1. Look in all places that text color can be specified
2. Check that text color is not specified
3. Look in all areas that background color or image used as a background can be specified
4. Check that no background color or image used as a background is specified
Expected Results: # 2 and 4 are true

#### Example
The author specifies neither text color nor background, and does not use CSS. As a result the user can set his browser defaults to provide the colors and contrasts that work well for them.

## Scenario: Alternative version
- [ ] Providing a control with a sufficient contrast ratio that allows users to switch to a presentation that uses sufficient contrast
 
A link or control on the page can either change the page so that all aspects conform, or it could take the viewer to a new version of the page that does conform at the desired level. Placing the link or control prominently on the page will assist users in accessing the conforming content readily.

**For this technique to be used successfully, three things must be true:**
1. The link or control on the original page must itself meet the contrast requirement of the desired SC. (If the user cannot see the control they may not be able to use it to go to the new page.)
2. The new page must contain all the same information and functionality as the original page.
3. The new page must conform to all of the SC for the desired level of conformance. (i.e., the new page cannot just have the desired level of contrast but otherwise not conform).

> NOTE
> This technique can be used in combination with a style switching technique to present a page that is a conforming alternate version for non-conforming content. Refer to [C29: Using a style switcher to provide a conforming alternate version](https://www.w3.org/WAI/WCAG21/Techniques/css/C29) and [Understanding Conforming Alternate Versions for more information]'https://www.w3.org/WAI/WCAG21/Understanding/conformance#conforming-alt-versions).

### Tests Procedure
1. Check that a link or control exists on the original page that provides access to the alternate version.
2. Check that the link or control on the original page conforms to all success criteria for the conformance level being tested.
3. Check that the 
Expected Results: All three checks are true.

#### Example
- A page with some headlines that do not meet the 3:1 contrast requirements has a high contrast (5:1) link at the top of the page that takes the user to a new version of the page with minimum 4.5:1 contrast on all text and images of text.
- A page uses shaded backgrounds for effect but results in text to background contrast of 4:1. A control at the top of the page says "high contrast". Clicking on it causes different styles to be used and dropping the background colors to achieve 7:1 contrast.

## Scenario:  Minimum contrast

- [ ] Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text
- [ ] For Large-scale text and images of large-scale text, ensuring that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text. **Large-scale**: with at least 18 point or 14 point bold or font size that would yield equivalent size for Chinese, Japanese and Korean (CJK) fonts

**Exception**
Text or images of text that are part of an inactive user interface component, that are pure decoration, that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.

### Tests Procedure
Suggested extension for test
- [Chrome extension](https://chrome.google.com/webstore/detail/wcag-color-contrast-check/plnahcmalebffmaghcpcmpaciebdhgdf)
- [Firefox extension](https://addons.mozilla.org/pt-BR/firefox/addon/wcag-contrast-checker/)

1. Check the contrast ratio in every page is equal to or greater than 
   1. 4.5:1 for small texts (check for level AA, small text in above extension)
   1. 3:1 for large texts (check for level AA, large text in above extension)
   
:x: Selected item failed check for level AA, small
![image](https://user-images.githubusercontent.com/993369/79802990-76ee7380-8337-11ea-928d-06ce30c0f2b7.png)
:x: Selected item failed check for level AA, large
![image](https://user-images.githubusercontent.com/993369/79803172-e06e8200-8337-11ea-9e05-9155b72fac32.png)

# Other tools for Contrast ratio checker / decision

- [Text on background image](https://www.brandwood.com/a11y)
- [Color safe palettes](http://colorsafe.co/)
- [Material.io color tool palette creation with accessibility validation](https://material.io/resources/color/#!/?view.left=1&view.right=0&primary.color=1E88E5&secondary.color=039BE5)
- Firefox's built-in developer tool has contrast checker
